### PR TITLE
Pornolab

### PR DESCRIPTION
**Pull request by lpearl**
Adding Pornolab support still not working tho.

### CODE REVIEW

**Comment by garfield69** [comment]
I've made some changes.
I think we are now at the point were we can login (with or without captcha) successfully, and return results.
The only thing I cant get to work is the download link, which for some reason fails because of the following reason:
```
<center><br /><br />Скачивание возможно только со страниц трекера.<br /><br />Попробуйте <a href="https://pornolab.net/forum/viewtopic.php?t=2594782">перейти на страницу с раздачей</a> и кликнуть на ссылку для скачивания торрент-файла.</center>
```
It is beyond my skills to fix :-(

**Comment by lpearl** [comment]
I'll take a look tomorrow but I'm also not the best at html and I'm still trying to figure out the file structure of jackett

**Comment by lpearl** [comment]
@garfield69 it seams like the Referer in the request header has to be spoofed. I know the website has a limit of the number of files you can download per day so as long as that number is still accounted for I don't think it will be a problem. I might be wrong but do you have any idea how to do this?

**Comment by garfield69** [comment]
here is an example:
https://github.com/Jackett/Jackett/blob/060972475f13ffe59fcc09c51ffe91a547a29029/src/Jackett.Common/Indexers/AnimeTorrents.cs#L24
https://github.com/Jackett/Jackett/blob/060972475f13ffe59fcc09c51ffe91a547a29029/src/Jackett.Common/Indexers/AnimeTorrents.cs#L123
if you cant figure out how to apply it to pornolab let me know and I'll have a go later.

**Comment by lpearl** [comment]
@garfield69 Been at it for awhile and still can’t figure it out. AnimeTorrents indexer uses a referer for the search not the download link. I can’t seam to figure how to add a referer to the link the download link. If it helps you the referer needs to be the same as the comments.

**Comment by garfield69** [comment]
Well that is a problem, because while the indexer sets up the url for the download, it does not actually action it.
When the user clicks on the prepared download url, its not the indexer that actions it, its the jackett standard download processor, which I think is https://github.com/Jackett/Jackett/blob/master/src/Jackett.Server/Controllers/DownloadController.cs
And I've no idea how, or even if its possible, to tell it to use a Referer.
I cant help further with this one, sorry. :-(

**Comment by lpearl** [comment]
Thanks for your help. I will see if I can get some sort of logic where if a referer is provided along with the URL it will use It. if it’s just the URL it will not. It will take me some time to figure out.

**Comment by adem4ik** [comment]
Just a random point:
Back in the days `pornolab` was forked from `rutracker` (ex-torrents.ru), and they are still more or less the similar. So I guess rutracker's implementation could be used as an example.

**Comment by garfield69** [comment]
@adem4ik yes that is true. in fact the pornolab indexer being developed here was based on a clone of the current RuTracker indexer source.
unfortunately the download process for the two sites is not the same any longer ;-(

**Comment by lpearl** [comment]
@garfield69 What is a public tracker that uses direct download so I can do some texting?

**Comment by garfield69** [comment]
Horrible Subs is the first public tracker I have found that has a Jackett Indexer written in C# and serves .torrent files as well as magnets. 
if that is no good for you let me know and I can keep looking for more.

**Comment by lpearl** [comment]
@garfield69 After doing some digging I found a download code used in [Torrentech](https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Indexers/Torrentech.cs) that seems to download using cookies (I think this would solve the problem) Its a little different and I'm not supper familiar with HTML you mind taking a look (code at line 319).

**Comment by garfield69** [comment]
Fantastic work.
I've tested this against Pornolan, rutracker, horriblesubs and 1337x and everything appears to be working well.
so unless there is more you want to do, say yay or nay to letting me accept the PR and merge.

**Comment by lpearl** [comment]
Yep, all good... one thing which does not matter but, just wondering is why there is a tv category? Other than that it looks good to merge. Any other c# trackers you want me to work on (I would like to help out and learn more)?
Screen Shot 2019-12-04 at 5 57 58 PM

File: src/Jackett.Common/Indexers/pornolab.cs [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+        {
+            get { return (ConfigurationDataPornolab)base.configData; }
+            set { base.configData = value; }
+        }
+
+        public Pornolab(IIndexerConfigurationService configService, WebClient wc, Logger l, IProtectionService ps)
+            : base(name: "Pornolab",
+                   description: "Pornolab is a Semi-Private Russian site for Adult content",
+                   link: "https://pornolab.net/",
+                   caps: TorznabUtil.CreateDefaultTorznabTVCaps(),
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/pornolab.cs]: >wondering why there is a tv category? 

https://github.com/Jackett/Jackett/blob/060972475f13ffe59fcc09c51ffe91a547a29029/src/Jackett.Common/Utils/TorznabCapsUtil.cs#L12-L21

**Comment by garfield69** [comment]
>Any other c# trackers you want me to work on (I would like to help out and learn more)?

take your pick ;-)
https://github.com/Jackett/Jackett/issues?q=is%3Aopen+is%3Aissue+label%3A%22needs+c%23%22

**Pull request merged**